### ISSUE TITLE
Archiving iOS App Bundle does not show full error message on failure

### ISSUE THREAD

**Issue by Mikilll94** [xamarin/xamarin-macios#7207]
### Steps to Reproduce

1. Archive app
2. Upload to App Store
3. An error occurs

```
Archiving App Bundle 'MyApp'...
Beginning build for D:\Projects\MyApp\MoTo\MoTo.iOS\MoTo.iOS.csproj. Properties: DeviceSpecificBuild: false, TargetiOSDevice: , ServerPort: 54886, ServerUser: sampleuser, ContinueOnDisconnected: False, ServerAddress: 172.23.136.187, ArchiveOnBuild: False, IsAppDistribution: False, ArchivePath: 
App archive 'MyApp' completed successfully.
Begining distribution, 10.10.2019 13:56:32
Channel: App Store
App Bundle Id: de.company.app, Version: 1.0
Publishing to 'C:\Users\sampleuser.OURNET\AppData\Local\Xamarin\MonoTouch\Archives\2019-10-10\MoTo.iOS 10-10-19 1.54 PM.xcarchive\Published\MyApp1.0.ipa'
Validating bundle...
Uploading 'C:\Users\sampleuser.OURNET\AppData\Local\Xamarin\MonoTouch\Archives\2019-10-10\MoTo.iOS 10-10-19 1.54 PM.xcarchive\Published\MyApp1.0.ipa'
Failed to upload MyApp bundle to the App Store
"altool" exited with code 176.

Publishing project 'MyApp' completed with errors.
Failed to upload MyApp bundle to the App Store
"altool" exited with code 176.
```

### Expected Behavior

App should be uploaded to App Store successfully

### Actual Behavior

A strange error pops up.

### Environment

```
Microsoft Visual Studio Professional 2019
Version 16.3.3
VisualStudio.16.Release/16.3.3+29403.142
Microsoft .NET Framework
Version 4.8.03761

Installed Version: Professional

Application Insights Tools for Visual Studio Package   9.1.00913.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2019   16.3.284.37798
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.3.284.37798
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.3.284.37798
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.3.284.37798
Azure Functions and Web Jobs Tools

C# Tools   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.0 (d16-2@8b56e20)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 0x10 - v2.9.20816.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.0.83+gbc8a4b23ec
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Mono Debugging for Visual Studio   16.3.7 (9d260c5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.61908.27190
Microsoft SQL Server Data Tools

Syntax Visualizer   1.0
An extension for visualizing Roslyn SyntaxTrees.

TypeScript Tools   16.0.10821.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.4 for F# 4.6   16.3.0-beta.19455.1+0422ff293bb2cc722fe5021b85ef50378a9af823
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.3.0.275 (d16-3@35ef585)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.3.0.246 (remotes/origin/d16-3@bd2f86892)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.3.565 (27e9746)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   10.0.3.0 (d16-3/4d45b41)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2019-06@5608fe0abb3
    Java.Interop: xamarin/java.interop/d16-3@5836f58
    LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3fd
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.27.1@8212a2d
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333


Xamarin.iOS and Xamarin.Mac SDK   13.2.0.47 (c2cbd34)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Build Logs

> VS bug [#1000580](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1000580)

**Comment by whitneyschmidt** [comment]
Thank you for your feedback!

This appears to be an issue related to notarization. Currently, Apple's notary service is down. You can check the system status here: https://developer.apple.com/system-status/

Have you received an email from Apple with more details about the failure?

**Comment by Mikilll94** [comment]
Ok. Finally, I have solved this problem. The reason for this error was uploading build with the same build number as the previous one. However, this error was so strange that I could not figure what is going on.

I came up with an idea when I have uploaded an app directly in terminal using `xcrun altool ...` When I did this I saw the right error message
```
"Redundant binary upload. You have uploaded an app with this bundle identifier
```
(or something like that)"

After unsuccessful upload to App Store I think we should present helpful error messages to the user because error
```
Failed to upload MyApp bundle to the App Store
"altool" exited with code 176.
```
is not helpful at all.
